### PR TITLE
fixing severe performance issue with ESmry

### PR DESCRIPTION
**Pull request by tskille**
This PR fixes a bug in ESmry which was introduced with PR #1092 (support for non-unified result files). The bug (which is now fixed) caused a severe performance regression on ESmry for unified result files.

### CODE REVIEW

File: src/opm/io/eclipse/ESmry.cpp [hunk 1]
```diff
@@ -286,12 +286,14 @@ ESmry::ESmry(const std::string &filename, bool loadBaseRunData)
         //    2 or 3 arrays pr time step.
         //       If timestep is a report step:  MINISTEP, PARAMS and SEQHDR
         //       else : MINISTEP and PARAMS
-        
-        
+
         size_t i = std::get<0>(arraySourceList[0]) == "SEQHDR" ? 1 : 0 ;
 
+        std::string prevFile = "";
```
Comment (akva2) [inline src/opm/io/eclipse/ESmry.cpp]: string is empty by default.

File: src/opm/io/eclipse/ESmry.cpp [hunk 2]
```diff
@@ -286,12 +286,14 @@ ESmry::ESmry(const std::string &filename, bool loadBaseRunData)
...
+        EclFile* pEclFile = nullptr;
```
Comment (akva2) [inline src/opm/io/eclipse/ESmry.cpp]: use std::unique_ptr. bare pointers are so 2010.

File: src/opm/io/eclipse/ESmry.cpp [hunk 3]
```diff
@@ -304,10 +307,21 @@ ESmry::ESmry(const std::string &filename, bool loadBaseRunData)
             
             i++;
 
-            EclFile resfile(std::get<1>(arraySourceList[i]));
+            if (std::get<1>(arraySourceList[i]) != prevFile){
+                
+                if (pEclFile != nullptr){
```
Comment (akva2) [inline src/opm/io/eclipse/ESmry.cpp]: you don't need this, it happens on itself if you assign a new value to the unique_ptr.

File: src/opm/io/eclipse/ESmry.cpp [hunk 4]
```diff
@@ -355,10 +369,11 @@ ESmry::ESmry(const std::string &filename, bool loadBaseRunData)
         fromReportStepNumber = toReportStepNumber;
 
         n--;
+        pEclFile.reset();
```
Comment (akva2) [inline src/opm/io/eclipse/ESmry.cpp]: will happen when it goes out of scope.

**Comment by akva2** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
> The bug

And what was the bug?

**Comment by tskille** [comment]
bug or bad design? The code did not scale to larger models. It worked on smaller cases but it used 20s to open the unified summary file for NORNE_ATW2013. After the fix, this is down to 0.5s. 

This part of the PR if fixing the issue. 

```C++
if (std::get<1>(arraySourceList[i]) != prevFile){
    pEclFile = std::make_unique<EclFile>(std::get<1>(arraySourceList[i]));
    pEclFile->loadData()
    prevFile = std::get<1>(arraySourceList[i]);
}    
```

The previous version was opening a new EclFile for each timestep. After the fix, this is done only when the file is changed.

**Comment by joakim-hove** [comment]
> The previous version was opening a new EclFile for each timestep. After the fix, this is done only when the file is changed.

OK - thanks.

**Comment by blattms** [comment]
backported to release via #1155

**Pull request merged**